### PR TITLE
Execute Batch Payload Operation via JDBC Receiver Adapter

### PR DESCRIPTION
**Pull request by I562950**
This is my first tutorial. Requesting review for publishing.

### CODE REVIEW

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 1]
```diff
@@ -0,0 +1,377 @@
+---
+title: Execute Batch Payload operations using Java Database Connectivity (JDBC) Receiver Adapter
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:2]: Please use title case

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+description: Learn how to connect, interact, and perform batch operations on your database via JDBC receiver adapter.
+keywords: tutorial
+auto_validation: true
+time: 40
+tags: [ tutorial>intermediate, software-product>sap-integration-suite]
+primary_tag: software-product>sap-business-technology-platform
+parser: v2
+---
+
+## You will learn
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:12]: Please say "How to"
Also remove tab stops

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+There are two types of Batch operations:
+
+-	Atomic: This mode considers each batch operation as a single unit. It updates the whole batch operation successfully or reverts the entire operation to its initial state if anything in the batch operation fails.
+
+-	Non-Atomic- The operation in this mode varies depending upon the database and its driver. Generally, it updates all the successfully executed records and throws an exception if anything fails. It does not revert the failed records to its initial state.
+
+Following the steps below, you will be able to setup a sample database, create an integration flow, upload drivers, and data sources required for JDBC receiver adapter. Then, you can create a payload, execute, and benefit from Batch Payload functionality.
+
+
+### Choose your Database
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:36]: please use sentence case

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+Based on your requirements, choose a database you wish to work on. JDBC receiver adapters supports more than ten databases. To know more, see [JDBC: Supported Databases](https://help.sap.com/docs/CLOUD_INTEGRATION/368c481cd6954bdfa5d0435479fd4eaf/88be64412f1b46d684dfba11f2767c5b.html?locale=en-US&version=Cloud).
+
+If you're connecting to on-premise database, then you must validate the connection in the **Cloud Connector**. For more information, see [JDBC: Access Control](https://help.sap.com/docs/CP_CONNECTIVITY/cca91383641e40ffbe03bdc78f00f681/f42fe4471d6a4a5fb09b7f3bb83c66a4.html?version=Cloud&locale=en-US).
+
+
+### Create data source and upload driver
+JDBC adapter supports third party and in-house databases on Neo and Cloud Foundry Business Technology environments. Before you establish the connection to these databases, or setup an integration flow, you need to upload drivers (applicable for HANA, ASE, PostgreSQL only) and add Data Sources.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:43]: Please fix grammar -> the setup, to set up

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+
+To execute this scenario, let's create a table with few columns.
+
+1. Add a content modifier and then add following query to its **Message Body**. To ensure duplicate row entry is restricted in your database, make one of the columns as primary key.
+
+    `create table samplejdbc (emp_id integer primary key, emp_name varchar(255));`
+
+      <!-- border -->![Add Content Modifier and input Query](Input-Query.png)
+
+2. Deploy this integration flow. We will now try to execute this via [Postman]( https://www.getpostman.com/) (You can use any other HTTPS client application too.)
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:78]: Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+ You can find the endpoint in **Manage Integration Content** section as shows below.
+
+    <!-- border -->![Find Endpoint](Find-Endpoint.png)
+
+    In the **Request** section, enter this **End-point** and **Send** the blank query. This will setup connection of Postman with your integration flow.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:83]: set up

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+
+    <!-- border -->![Send query from Postman](Blank-Query-Send.png)
+
+3. Before proceeding to the next step, remove Content Modifier from your integration flow. We had added it to create a database successfully.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:87]: Is Content Modifier a product name? If not, please use lower case or put it in back-ticks.
Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+
+
+
+> For simplicity, we will use Postman to push the payload, later we will also see how to use Groovy Script to push the payload.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:91]: Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 9]
```diff
@@ -0,0 +1,377 @@
...
+
+
+### Execute Batch Payload
+
+1. In your integration flow, edit the settings for JDBC Receiver adapter. Under **Connections** tab, select the **Batch Payload** checkbox and **Batch Operation** as **Atomic**. We are doing this because Batch supports XML payload in INSERT, UPDATE and DELETE modes only. Native SQL queries are supported with prepared statements only.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:96]: Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+
+    <!-- border -->![Batch Payload Settings](Batch-Payload-Setting.png)
+
+2. Open your HTTP client (in this case Postman) to create a payload. Enter the **Username** and **Password** under **Authorisation** tab.
+
+    <!-- border -->![Postman- Authorization](Postman-Authorization.png)
+
+3. To create Payload, we will use INSERT mode and create a payload (with table name `samplejdbctest` ) to test Batch Payload scenario. To use these modes, you must follow the syntax. **Send** the following payload from **Request > Body** tab of postman:
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:104]: Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 11]
```diff
@@ -0,0 +1,377 @@
...
+> Ensure to check the time stamp to determine which record to check.
+
+
+
+### Atomic Batch Operation
+
+Atomic Batch Operation doesn't allow an action to be performed on the payload if an error is encountered. To verify that, follow these steps:
+
+
+1. Copy the above payload and modify just one record and make it unique. For example, here we are making the **`emp_id`** (of second record) as 3. Thus, making that record unique, but the first record is duplicate. Send this query.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:148]: Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 12]
```diff
@@ -0,0 +1,377 @@
...
+
+    <!-- border -->![Batch Payload Settings](Batch-Payload-Setting.png)
+
+2. Open your HTTP client (in this case Postman) to create a payload. Enter the **Username** and **Password** under **Authorisation** tab.
+
+    <!-- border -->![Postman- Authorization](Postman-Authorization.png)
+
+3. To create Payload, we will use INSERT mode and create a payload (with table name `samplejdbctest` ) to test Batch Payload scenario. To use these modes, you must follow the syntax. **Send** the following payload from **Request > Body** tab of postman:
+
+    ```
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:106]: Please add keyword for code highlighting

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 13]
```diff
@@ -0,0 +1,377 @@
...
+
+
+### Atomic Batch Operation
+
+Atomic Batch Operation doesn't allow an action to be performed on the payload if an error is encountered. To verify that, follow these steps:
+
+
+1. Copy the above payload and modify just one record and make it unique. For example, here we are making the **`emp_id`** (of second record) as 3. Thus, making that record unique, but the first record is duplicate. Send this query.
+
+    ```
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:150]: Please add keyword for code highlighting

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 14]
```diff
@@ -0,0 +1,377 @@
...
+
+Then, **Send** the following query from Postman:
+
+`select * from samplejdbctest`
+
+  <!-- border -->![Atomic Batch Operation Verification](Verify-Atomic.png)
+
+This response shows that no new data has been inserted. Hence, verifying the Atomic operation.
+
+### Non-Atomic Batch Operation
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:193]: Please use sentence case

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 15]
```diff
@@ -0,0 +1,377 @@
...
+
+To check the Message Processing Log (MPL) of this integration flow from your tenant, go to **Monitoring** > **Overview** > **All Integration flows**. Choose the latest record with your artefact name.
+
+  <!-- border -->![Payload Processed](Payload-Processed.png)
+
+> Ensure to check the time stamp to determine which record to check.
+
+
+
+### Atomic Batch Operation
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:143]: Please use sentence case

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 16]
```diff
@@ -0,0 +1,377 @@
...
+### Non-Atomic Batch Operation
+
+
+Depending upon your driver (which depends upon choice of database), the behaviour of Non- Atomic mode differs. Say for SQL server database, you would be able to insert the unique record and it would fail all the duplicate records. So, the complete payload doesn't fail. But for some other database and driver, it may act similar to Atomic mode and fail the payload even if single error is encountered.
+
+1. In your integration flow, edit the settings for JDBC Receiver adapter. Under **Connections** tab, select the **Batch Payload** checkbox and select **Batch Operation** as Non-Atomic.
+
+    <!-- border -->![Batch Payload Setting Non-Atomic](Batch-Payload-Setting-Non-Atomic.png)
+
+2. **Send** this payload from Postman. Here we are making the **`emp_id`** (of second record) as 4. Thus, making that record unique, but the first record is duplicate.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:202]: Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 17]
```diff
@@ -0,0 +1,377 @@
...
+4. **Send** the following query from Postman:
+    `select * from samplejdbctest`  
+
+      <!-- border -->![Non-Atomic Batch Mode Verification](Verify-Non-Atomic.png)
+
+So, it is evident that using Non-Atomic mode, the new record is getting inserted and the duplicate record fails. So, the whole payload doesn't fail unlike Atomic Batch operation.
+
+>The function/behaviour of Non-Atomic Batch Operation may depend upon the choice of database.
+
+### Send Payload using Groovy Script
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:245]: Sentence case

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 18]
```diff
@@ -0,0 +1,377 @@
...
+
+You can also upload the payload using the **Groovy Script**. In this case you must use the prepared statements.
+
+
+1. Uncheck **Batch Mode** in the integration flow.
+<!-- border -->![Batch Mode Off](Batch-Off.png)
+Let's insert a new database table. We have already seen creating sample database via **Content Modifier**. Now we will do it directly from Postman. **Send** this query from Postman.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:252]: Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 19]
```diff
@@ -0,0 +1,377 @@
...
+
+Depending upon your driver (which depends upon choice of database), the behaviour of Non- Atomic mode differs. Say for SQL server database, you would be able to insert the unique record and it would fail all the duplicate records. So, the complete payload doesn't fail. But for some other database and driver, it may act similar to Atomic mode and fail the payload even if single error is encountered.
+
+1. In your integration flow, edit the settings for JDBC Receiver adapter. Under **Connections** tab, select the **Batch Payload** checkbox and select **Batch Operation** as Non-Atomic.
+
+    <!-- border -->![Batch Payload Setting Non-Atomic](Batch-Payload-Setting-Non-Atomic.png)
+
+2. **Send** this payload from Postman. Here we are making the **`emp_id`** (of second record) as 4. Thus, making that record unique, but the first record is duplicate.
+
+    ```
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:204]: Please add keyword for code highlighting

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 20]
```diff
@@ -0,0 +1,377 @@
...
+
+    `create table testjdbcsample (emp_id integer primary key, emp_name varchar (255), join_date varchar(255), a1 integer, a2 integer, email varchar(255));`
+
+      <!-- border -->![Create database from Postman](Database-Postman.png)
+
+2. In the integration flow, insert the **Groovy script** to pass the payload and click **Select**.
+
+    <!-- border -->![Insert Groovy Script](Groovy-Script.png)
+
+    ```
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:262]: Please add keyword for code highlighting

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 21]
```diff
@@ -0,0 +1,377 @@
...
+        message.setHeader("CamelSqlParameters",paramList);
+
+        //Body
+        message.setBody("INSERT INTO testjdbcsample (emp_id,emp_name,join_date,a1,a2,email) VALUES(?,?,?,?,?,?)");
+        return message;
+    }
+    ```
+3. Select the **Batch Mode** checkbox and click **Deploy**.
+
+4. Run a blank query in Postman to send the payload which we have entered via Groovy Script and establish a connection between Postman and your integration flow.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:285]: Don't say "we" when addressing the author. Say "you" instead!
Missing comma before "which"

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 22]
```diff
@@ -0,0 +1,377 @@
...
+    <!-- border -->![Send Payload from Postman for Groovy Script](Postman-Groovy.png)
+
+5. Let's verify if the payload has been successfully sent to the database. Remove **Groovy Script**, deselect the **Batch Mode** and **Deploy** the integration flow. Then, run the following query from Postman:
+`select * from testjdbcsample`
+    <!-- border -->![Link text e.g., Destination screen](Postman-Groovy-Success.png)
+
+This displays the table data. Hence, we have successfully sent a payload via Groovy Script. You can now alter the table data and verify the Atomic and Non Atomic behaviours as we did earlier.
+
+### Atomic Batch Operation with Groovy Script
+We will replicate the above Atomic scenario with a Groovy Script.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:295]: Don't say "we" when addressing the author. Say "you" instead!

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 23]
```diff
@@ -0,0 +1,377 @@
...
+4. Run a blank query in Postman to send the payload which we have entered via Groovy Script and establish a connection between Postman and your integration flow.
+    <!-- border -->![Send Payload from Postman for Groovy Script](Postman-Groovy.png)
+
+5. Let's verify if the payload has been successfully sent to the database. Remove **Groovy Script**, deselect the **Batch Mode** and **Deploy** the integration flow. Then, run the following query from Postman:
+`select * from testjdbcsample`
+    <!-- border -->![Link text e.g., Destination screen](Postman-Groovy-Success.png)
+
+This displays the table data. Hence, we have successfully sent a payload via Groovy Script. You can now alter the table data and verify the Atomic and Non Atomic behaviours as we did earlier.
+
+### Atomic Batch Operation with Groovy Script
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:294]: sentence case

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 24]
```diff
@@ -0,0 +1,377 @@
...
+We will replicate the above Atomic scenario with a Groovy Script.
+
+
+1. Choose **Atomic Batch Mode** and add the following script in Groovy Script as explained in Step 8.
+
+    Here we have modified the third and fourth record of the original Groovy Script.
+
+    ```
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:302]: Please add keyword for code highlighting

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 25]
```diff
@@ -0,0 +1,377 @@
...
+            paramList.add(Arrays.asList(112, 'test112', 2021-01-01,1,1,'test@gmail.com'));
+            paramList.add(Arrays.asList(106, 'test106', 2021-01-01,1,1,'test@gmail.com'));
+            message.setHeader("CamelSqlParameters",paramList);
+
+            //Body
+            message.setBody("INSERT INTO testjdbcsample (emp_id,emp_name,join_date,a1,a2,email) VALUES(?,?,?,?,?,?)");
+            return message;
+        }
+    ```  
+2. **Send** a blank query in Postman to send the payload which we have entered via Groovy Script.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:323]: Don't say "we" when addressing the author. Say "you" instead!
Missing comma before "which"
Check upper/lower case-> "Script"

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 26]
```diff
@@ -0,0 +1,377 @@
...
+4. Remove **Groovy Script**, switch off the **Batch Mode** and **Deploy** the integration flow. Then, run the following query from Postman to check the database updates, if any:
+
+`select * from testjdbcsample`
+
+  <!-- border -->![Link text e.g., Destination screen](Groovy-Atomic-Error.png)
+
+This proves the Atomic mode functionality as the database table is not updated. It is the same as it was after Step 8. The new record didn't insert because there were duplicate records.
+
+
+### Non-Atomic Batch Operation with Groovy Script
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:338]: sentence case

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 27]
```diff
@@ -0,0 +1,377 @@
...
+
+1. Choose **Non-Atomic Batch Mode** and add the following script in Groovy Script as explained in Step 8.
+
+    ```
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:342]: Please add keyword for code highlighting

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md [hunk 28]
```diff
@@ -0,0 +1,377 @@
...
+        paramList.add(Arrays.asList(112, 'test112', 2021-01-01,1,1,'test@gmail.com'));
+        paramList.add(Arrays.asList(106, 'test106', 2021-01-01,1,1,'test@gmail.com'));
+        message.setHeader("CamelSqlParameters",paramList);
+
+        //Body
+        message.setBody("INSERT INTO testjdbcsample (emp_id,emp_name,join_date,a1,a2,email) VALUES(?,?,?,?,?,?)");
+        return message;
+    }
+    ```
+2. **Send** a blank query in Postman to send the payload which we have entered via Groovy Script.
```
Comment (MichaelCzcz) [inline tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:363]: Don't say "we" when addressing the author. Say "you" instead!
Missing comma before "which"

**Review by MichaelCzcz — CHANGES_REQUESTED** [review]
Thank you for creating a new tutorial.
Please address my comments and proof read the whole tutorial, because there are some names/phrases where it is unclear if the names/phrases are product names and should start with an upper case letter, for example Groovy Script and Batch Operation.
There are also some images where the canvas is bigger than the actual content. Please fix this.

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 1 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:2]: Done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 3 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:36]: The words, JDBC, Batch Payload have to be in title case. These are product names/functionalities.

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 4 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:43]: reframed

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 5 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:78]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 6 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:83]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 7 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:87]: yes, it is a product name.

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 8 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:91]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 9 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:96]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 10 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:104]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 11 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:148]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 14 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:193]: Atomic is a special functionality of the product (JDBC adapter), which we are demonstrating here.

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 15 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:143]: done, Atomic Batch operation.

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 17 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:245]: done
Send payload using Groovy Script

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 18 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:252]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 21 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:285]: done and reframed

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 22 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:295]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 23 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:294]: Atomic Batch operation with Groovy Script

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 25 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:323]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 26 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:338]: Non-Atomic Batch operation with Groovy Script

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 12 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:106]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 13 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:150]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 19 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:204]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 20 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:262]: The whole code has to be inserted in Groovy Script.

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 24 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:302]: done

File: tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md (hunk 27 above)
Comment (I562950) [reply tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md:342]: done

**Comment by I562950** [comment]
> Have you published this tutorial to the QA system? I cannot find it there and I prefer to view it in the QA system before merging.

I have published it, but it has not appeared on QA yet. https://developers-qa.sap.com/tutorials/btp-integration-suite-execute-batch-payload.html

**Comment by jmmargo** [comment]
I did see that it was published on QA Blue: https://developers-qa-blue.wcms-nonprod.c.eu-de-2.cloud.sap/tutorials/btp-integration-suite-execute-batch-payload.html.  I will review it this morning.

**Pull request merged**